src/data/battler-tags.ts

```diff
@@ -2244,7 +2318,9 @@ export function getBattlerTag(tagType: BattlerTagType, turnCount: number, source
   case BattlerTagType.IGNORE_FLYING:
     return new GroundedTag(tagType, BattlerTagLapseType.CUSTOM, sourceMove);
   case BattlerTagType.ROOSTED:
-    return new GroundedTag(tagType, BattlerTagLapseType.TURN_END, sourceMove);
+    return new RoostedTag(tagType, BattlerTagLapseType.TURN_END, sourceMove);
+  case BattlerTagType.BURNED_UP:
+    return new BurnedUpTag(tagType, BattlerTagLapseType.CUSTOM, sourceMove);
```

Comment:
```suggestion
    return new RoostedTag();
  case BattlerTagType.BURNED_UP:
    return new BurnedUpTag();
```
To go along with other suggestions...

Also, doesn't Double Shock also need a tag in this vein?

Reply:
Ill take a peek at this, hopefully its the same as what i did for burn up, will need another test case tho